### PR TITLE
Fix #2051 wrong is_member() negation

### LINKED ISSUE #2051: Wrong is_member() negation
As I mentioned in another PR, this check shouldn't be negated: https://github.com/mybb/mybb/blob/feature/inc/datahandlers/post.php#L696 - `$num_prefixes` should be `true` only if the user can use a prefix. And the `-1` check is not needed anymore either.

### PR DESCRIPTION
**Pull request by Destroy666x**
Fixes one invalid `is_member()` negation, returns empty strings instead
of `false` for HTML code, shortens redundant code (`-1` returns always
`true` for `is_member()` and `get_user()` the current user - `$mybb->user` - if the UID
matches).

https://github.com/mybb/mybb/issues/2051

### CODE REVIEW

File: inc/datahandlers/post.php [hunk 1]
```diff
@@ -708,14 +708,7 @@ function verify_prefix()
...
 
-					if($required['groups'] != "-1")
-					{
-						if(!is_member($required['groups'], array('usergroup' => $user['usergroup'], 'additionalgroups' => $user['additionalgroups'])))
-						{
-							$num_prefixes = true;
-						}
-					}
-					else
+					if(is_member($required['groups'], array('usergroup' => $user['usergroup'], 'additionalgroups' => $user['additionalgroups'])))
```
Comment (JN-Jones) [inline inc/datahandlers/post.php]: As I've already mentioned elsewhere: Why not pass `$user` directly? (IIRC I've already change it here in one of my PRs)

Comment (Destroy666x) [inline inc/datahandlers/post.php]: I can change it, but I don't think it makes a big difference. This way clearly shows that groups are passed and the function only receives the values it needs.

Comment (JN-Jones) [inline inc/datahandlers/post.php]: Imho it's easier to read: it checks whether `$user` is in `$required['groups']`

**Comment by JN-Jones** [comment]
Produced merge conflicts with the doc blocks.

**Comment by JN-Jones** [comment]
@Destroy666x can you fix the conflicts? ;)

**Comment by JN-Jones** [comment]
@Destroy666x

**Comment by Destroy666x** [comment]
Will try to fix 2-5 of my assigned issues a day starting this weekend.

**Comment by Destroy666x** [comment]
Conflicts were fixed, somehow

**Pull request merged**